plutus-tx-plugin/src/Language/PlutusTx/Plugin.hs

```diff
@@ -312,14 +283,15 @@ runCompiler opts expr = do
     let ctx = PIR.defaultCompilationCtx
               & set (PIR.ccOpts . PIR.coOptimize) (poOptimize opts)
               & set PIR.ccBuiltinMeanings PLC.getStringBuiltinMeanings
-              & set PIR.ccTypeCheckConfig (PIR.PirTCConfig stringBuiltinTCConfig PIR.YesEscape)
+              & set PIR.ccTypeCheckConfig (
+                  if poDoTypecheck opts
+                  then Just $ PIR.PirTCConfig stringBuiltinTCConfig PIR.YesEscape
+                  else Nothing)
```

Comment:
IMO this is an example of why the change you made is maybe not a good thing. Rather than being able to just set the typechecking config that we will need if we are going to typecheck, we also have to work in the decision about whether to typecheck or not.

At any rate, the usual rule about multiline expressions suggests putting this in a variable above.